### PR TITLE
Multiple chain metadata values

### PR DESCRIPTION
**Pull request by jonmattgray**
Having multiple metadata values will allow us to separate out the JSON blob from the base64 encoded content file in the demo. This will allow someone to understand the meaning of a token without needing to download the attached file.

- [x] Multiple metadata (key, value) pairs against a token
- [x] Update tests
- [x] Remove old migration 
- [x] Configurable and enforced maximum number of metadata items supported by the pallet
- [x] Update codeowners
- [x] Minor version bump -> `2.1.0`
- [x] Update README
- [x] Regenerate weights for run_process

### CODE REVIEW

File: pallets/simple-nft/src/lib.rs [hunk 1]
```diff
@@ -109,7 +111,7 @@ pub mod pallet {
         pub(super) fn run_process(
```
Comment (mattdean-digicatapult) [inline pallets/simple-nft/src/lib.rs:111]: need to regenerate weights for `run_process`. I think this is described inthe readme 🤔

File: pallets/simple-nft/src/tests.rs [hunk 2]
```diff
@@ -320,51 +344,77 @@ fn it_fails_for_destroying_multiple_tokens_as_other_signer() {
...
         assert_eq!(token_1, SimpleNFTModule::tokens_by_id(1));
-        // asset old token hasn't changed
+        // assert old token hasn't changed
         assert_eq!(token_2, SimpleNFTModule::tokens_by_id(2));
     });
 }
+
+
+#[test]
+fn it_fails_for_creating_single_token_with_too_many_metadata_items() {
```
Comment (mattdean-digicatapult) [inline pallets/simple-nft/src/tests.rs:403]: Need another test covering multiple metadata values on a token and perhaps one where multiple outputs have different numbers of valid metadata items

File: runtime/src/lib.rs [hunk 3]
```diff
@@ -272,12 +272,18 @@ impl pallet_node_authorization::Config for Runtime {
...
+parameter_types! {
+    pub const MaxMetadataCount: u32 = 16;
+}
+
 /// Configure the template pallet in pallets/simple-nft.
 impl pallet_simple_nft::Config for Runtime {
     type Event = Event;
     type TokenId = u128;
-    type TokenMetadata = Hash;
+    type TokenMetadataKey = Vec<u8>;
```
Comment (mattdean-digicatapult) [inline runtime/src/lib.rs:283]: I wonder if this would be better as a length limited array, `[u8; 32]` say

**Review by mattdean-digicatapult — CHANGES_REQUESTED** [review]
Few minor issues. Looking good

File: pallets/simple-nft/src/tests.rs (hunk 2 above)
Comment (jonmattgray) [reply pallets/simple-nft/src/tests.rs:493]: Added a multiple metadata values on a single token test and I think I've added the second one?

Comment (jonmattgray) [reply pallets/simple-nft/src/tests.rs:493]: https://github.com/digicatapult/vitalam-node/pull/8/commits/305ff62fd14cc0ac3e0a7e6cf5a5be35589dd5a9

File: runtime/src/lib.rs (hunk 3 above)
Comment (jonmattgray) [reply runtime/src/lib.rs]: changed, still `u64` for tests

Comment (jonmattgray) [reply runtime/src/lib.rs]: and updated API to handle padding

File: pallets/simple-nft/src/lib.rs (hunk 1 above)
Comment (jonmattgray) [reply pallets/simple-nft/src/lib.rs:111]: done

**Review by darrylmorton-digicatapult — APPROVED** [review]

File: pallets/simple-nft/src/weights.rs [hunk 4]
```diff
@@ -27,18 +27,16 @@ pub trait WeightInfo {
...
+	fn run_process(i: usize, o: usize) -> Weight {
+		(104_267_000 as Weight)
+			// Standard Error: 6_647_000
+			.saturating_add((13_768_000 as Weight).saturating_mul(i as Weight))
+			.saturating_add(T::DbWeight::get().reads(1 as Weight))
+			.saturating_add(T::DbWeight::get().reads((1 as Weight).saturating_mul(i as Weight)))
+			.saturating_add(T::DbWeight::get().writes(1 as Weight))
+			.saturating_add(T::DbWeight::get().writes((1 as Weight).saturating_mul(i as Weight)))
+			.saturating_add(T::DbWeight::get().writes((1 as Weight).saturating_mul(o as Weight)))
+	}
```
Comment (mattdean-digicatapult) [inline pallets/simple-nft/src/weights.rs:39]: This is suspicious. Specifically that the base weight has gone up by so much and there is now no element based on processing the each output like ` .saturating_add((8_875_000 as Weight).saturating_mul(o as Weight))`. I think we can let this pass, but we should look into improving these as time goes on

**Review by mattdean-digicatapult — APPROVED** [review]
LGTM with one comment for the future. Can you create a ticket to re-evaluate our weights and benchmarks once the process abstraction is done?

**Pull request merged**